crates/provider/src/builder.rs

```diff
@@ -225,23 +226,25 @@ impl<L, N> ProviderBuilder<L, N> {
         Ok(self.on_client(client))
     }
 
+    /// Build this provider with an Reqwest HTTP transport.
     #[cfg(feature = "reqwest")]
     pub fn on_reqwest_http(self, url: url::Url) -> Result<L::Provider, TransportError>
     where
         L: ProviderLayer<RootProvider<N, BoxTransport>, N, BoxTransport>,
```

Comment:
this is probably on me, but we should not box here, given the user has explicitly specified the transport type to be used. replace these `BoxTransport` usages with naming the Http type